### PR TITLE
@invertase/refactor import script

### PR DESCRIPTION
**Pull request by cabljac**
This PR refactors the import script code. 

TODO:
- full testing of the script
- update docs to reflect the fact that multi-thread is only supported for collection groups currently (not degradation, this was always the case)

### CODE REVIEW

File: firestore-bigquery-export/guides/IMPORT_EXISTING_DOCUMENTS.md [hunk 1]
```diff
@@ -17,11 +17,15 @@ You may pause and resume the import script from the last batch at any point.
...
+- You can use wildcard notation in the collection path. Suppose, for example, you have collections `users/user1/pets` and `users/user2/pets`, but also `admins/admin1/pets`. If you set `${COLLECTION_GROUP_QUERY}` to `true` and provide the collection path as `${users/{uid}/pets}`, the import script will import the former two collections but not the later, and will populate the `path_params` column of the big query table with the relevant `uid`s.
+
+- You can also use a [collectionGroup](https://firebase.google.com/docs/firestore/query-data/queries#collection-group-query) query. To use a `collectionGroup` query, provide the collection name value as `${COLLECTION_PATH}`, and set `${COLLECTION_GROUP_QUERY}` to `true`. For example, if you are trying to import `/collection/{document}/sub_collection`, the value for the `${COLLECTION_PATH}` should be provided as `sub_collection`. Keep in mind that if you have another sub collection with the same name (e.g. `/collection2/{document}/sub_collection`, that will be imported too.
```
Comment (cabljac) [inline firestore-bigquery-export/guides/IMPORT_EXISTING_DOCUMENTS.md:22]: This doc needs updating to mention the multithread support, and also clarify for collection group queries

File: firestore-bigquery-export/scripts/import/src/run-multi-thread.ts [hunk 2]
```diff
@@ -0,0 +1,111 @@
+import * as firebase from "firebase-admin";
+import { cpus } from "os";
+import { pool } from "workerpool";
+
+import * as logs from "./logs";
+import { CliConfig } from "./types";
+
+// import {
```
Comment (dackers86) [inline firestore-bigquery-export/scripts/import/src/run-multi-thread.ts:8]: Is this no longer needed?

File: firestore-bigquery-export/scripts/import/src/run-multi-thread.ts [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+//   dataSink: FirestoreBigQueryEventHistoryTracker
+// ): Promise<number> {
+//   const {
+//     datasetId,
+//     datasetLocation,
+//     rawChangeLogName,
+//   } = config;
+
+//   return processCollectionGroup(config)
+// }
```
Comment (dackers86) [inline firestore-bigquery-export/scripts/import/src/run-multi-thread.ts:111]: Is this comment no longer needed?

Comment (cabljac) [reply firestore-bigquery-export/scripts/import/src/run-multi-thread.ts:111]: will remove, no longer needed

**Review by dackers86 — APPROVED** [review]

**Pull request merged**